### DISCUSSION TITLE
How to remap ALT+SHIFT

### DISCUSSION THREAD

**Discussion by dinkonin** [xremap/xremap#392] [Q&A]
Is it possible to remap pressing ALT-SHIFT to META+SPACE? This works in keyd 
```ini
[alt]
shift = M-space
```
but I cannot make it work in xremap.

**Comment by k0kubun** [comment]
In xremap, modifiers don't finalize a key combination unless it's remapped to a multi-purpose (`alone`/`held`) key, like how they are usually handled by operating systems. For example, you may remap `shift` to `alone: something` and `held: shfit_L` in `modmap`, and then remap `Alt-something` to `M-space` in `keymap`, and that's possible in xremap too.

Without that design, the behavior of modifiers would be kind of complicated. For example, if you remap both `Alt-Shift` and `Alt-Shift-something` to two different things with keyd, I can't imagine what happens when you press `Alt-Shift-something`. With the above example for xremap, it's decided by how you press `Shift` as you configured.

**Comment by hpccc53** [comment]
Same as: #610
